File: solutions/2404/2404.java
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.HashMap;

class Solution {
    public int mostFrequentEven(int[] nums) {
        HashMap<Integer, Integer> evenNumbers = new HashMap<>();
        int maxFrequency = 0;
        ArrayList<Integer> maxFrequentNumbers = new ArrayList<Integer>();
        for (int element : nums) {
            Integer number = element;
            if (element % 2 != 0) continue;
            if (!evenNumbers.containsKey(number)) evenNumbers.put(number, 0);
            int frequency = evenNumbers.get(number) + 1;
            evenNumbers.put(number, frequency);
            if (maxFrequency < frequency) {
                maxFrequency = frequency;
                maxFrequentNumbers.clear();
            }
            if (maxFrequency == frequency) {
                maxFrequentNumbers.add(number);
            }
        }
        if (maxFrequency == 0) return -1;
        return Collections.min(maxFrequentNumbers);
    }
}

File: solutions/2404/2404.js
/**
* @param {number[]} nums
* @return {number}
*/
var mostFrequentEven = function(nums) {
  const evenNumsMap = new Map()
  let maxFrequency = 0
  let maxFrequentNumbers = new Array()
  
  for (const element of nums) {
    if (element % 2 != 0) continue
    if (!evenNumsMap.has(element)) {
      evenNumsMap.set(element, 0)
    }
    let frequency = evenNumsMap.get(element) + 1
    evenNumsMap.set(element, frequency)
    if (maxFrequency < frequency) {
      maxFrequency = frequency
      maxFrequentNumbers.length = 0
    }
    if (maxFrequency == frequency) {
      maxFrequentNumbers.push(element)
    }
  }
  if (maxFrequency == 0) return -1
  return Math.min(...maxFrequentNumbers)
};